modules/core/include/opencv2/core/mat.inl.hpp

```diff
@@ -1096,6 +1096,12 @@ void Mat::push_back(const Mat_<_Tp>& m)
     push_back((const Mat&)m);
 }
 
+template<> inline
```

Comment:
It's needed in order to define it as a specialization of this method:
template<typename _Tp> void push_back(const _Tp& elem);

Omitting template<> without adding an explicit overload in the class definition results in the compilation error:
prototype for 'void cv::Mat::push_back(const cv::MatExpr&)' does not match any in class 'cv::Mat' candidates are...

So we either add a specialization to the template overload of push_back() or add a new overload.